src/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -2268,6 +2268,10 @@ internal void AdjustScale(int digits, bool fRound)
 
             bNewScale = (byte)(lAdjust + _bScale);
             bNewPrec = (byte)(Math.Min(s_NUMERIC_MAX_PRECISION, Math.Max(1, lAdjust + _bPrec)));
+            if (Value == Decimal.Zero && _bPrec == 1 && _bScale == 0 && lAdjust > 0)
```

Comment:
Also, can you add a comment explaining why this edge case is necessary? Maybe something like this:

// Check for edge case with plain zero. "0" has scale=0 & precision=1, but parsed values of 
// zero with zeroes after the decimal (e.g. "0.000") have scale equal to their precision.